### PR TITLE
Token Gated Parties

### PR DESCRIPTION
**Pull request by anna-carroll**

### CODE REVIEW

File: contracts/bid/PartyBid.sol [hunk 1]
```diff
@@ -100,6 +105,9 @@ contract PartyBid is Party {
         nftContract = IERC721Metadata(_nftContract);
         tokenId = _tokenId;
         auctionId = _auctionId;
+        // TODO: validate that gatedToken is a contract
```
Comment (steveklebanoff) [inline contracts/bid/PartyBid.sol:108]: would it be worth calling an erc-20 specific fn call to ensure it's not just a contract but also an erc20?

**Review by steveklebanoff — COMMENTED** [review]
LGTM pending nit & at least one test :)

File: contracts/bid/PartyBid.sol [hunk 2]
```diff
@@ -38,6 +38,9 @@ import {
     IERC721Metadata
 } from "@openzeppelin/contracts/token/ERC721/extensions/IERC721Metadata.sol";
 import {IWETH} from "../external/interfaces/IWETH.sol";
+import {
```
Comment (steveklebanoff) [inline contracts/bid/PartyBid.sol:41]: Can we update the version number since this is a new ABI?

**Review by steveklebanoff — CHANGES_REQUESTED** [review]

File: contracts/bid/PartyBid.sol (hunk 2 above)
Comment (anna-carroll) [reply contracts/bid/PartyBid.sol:41]: this will be included in the PartyBuy release, version number is updated in that PR

**Pull request merged**